### PR TITLE
Fix/gulp css

### PR DESCRIPTION
**Pull request by tofixx**
- seems, that #854 resulted in failures compiling the scss files. please check, `gulp clear && gulp` will result now in compiling all files correctly
- missing return statements reuired to detect when tasks finished in gulp were added too.
- first run detection has been updated after styles task

image

### CODE REVIEW

**Comment by schurigh** [comment]
I didn't had any CSS problems the last days. May someone review who had.

File: static/styles/lib/base.scss [hunk 1]
```diff
@@ -4,7 +4,7 @@
 @import "./tether/tether.min.css";
 @import "./bootstrap/scss/bootstrap-flex";
 $fa-font-path: "/vendor-optimized/font-awesome/fonts";
-@import "font-awesome/scss/font-awesome.scss";
+@import "../../../node_modules/font-awesome/scss/font-awesome.scss";
```
Comment (Michael-Janke) [inline static/styles/lib/base.scss]: ../../ is not a proper way to get to node_modules
why changing this?

Comment (Michael-Janke) [reply static/styles/lib/base.scss]: try
@import "~font-awesome/scss/font-awesome";
(i have no time sry)

Comment (tofixx) [reply static/styles/lib/base.scss]: it should if webpack would be used for that, tried some variants but they didn't work.

Comment (Michael-Janke) [reply static/styles/lib/base.scss]: You are right!
https://stackoverflow.com/questions/41647984/importing-sass-partials-from-node-modules
with importPaths it should work

@adrianjost maybe you can doublecheck

File: gulpfile.js [hunk 2]
```diff
@@ -91,14 +91,13 @@ sassGrapher.init('./static/styles/', {
...
   var themeFile = `./theme/${themeName()}/style.scss`;
-  beginPipe('./static/styles/**/*.{css,sass,scss}')
+  return beginPipe('./static/styles/**/*.{css,sass,scss}')
     .pipe(gulpif(!firstRun, sassGrapher.ancestors()))
     .pipe(header(fs.readFileSync(themeFile, 'utf8')))
     .pipe(filelog("PROCESS: "))
     .pipe(sourcemaps.init())
     .pipe(sass({
-      sourceMap: true,
-      includePaths: ['node_modules']
```
[the remark this answers is not in the document]
Comment (tofixx) [reply gulpfile.js]: the initial issue was, that the css files were not build without the output of any error message. not sure how it's working now :)

**Pull request merged**